Add gitignore syntax highlighting

SPDX-License-Identifier: GPL-3.0-or-later

=================================================
diff -urN a/runtime/syntax/gitignore.yaml b/runtime/syntax/gitignore.yaml
--- a/runtime/syntax/gitignore.yaml	1970-01-01 00:00:00.000000000 +0000
+++ b/runtime/syntax/gitignore.yaml	2024-11-01 20:06:29.687293278 +0000
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+filetype: gitignore
+
+detect:
+  filename: "\\.gitignore$"
+
+rules:
+  - preproc: "\\[[^\\]]*\\]"
+  - symbol.operator: "\\*|\\!|\\?"
+  - symbol.brackets: "\\[|\\]"
+  - comment.line:
+      start: "#"
+      end: "$"
+      rules: []
